effects/gang_effect.py

```diff
@@ -0,0 +1,52 @@
+import message_text_config as msg
+import utils
+from citizens.citizen import Citizen
+from citizens.player import Player
+from game import Game
+
+from effects.effect import Effect, EffectStatus, InputStatusCode
+
+
+class GangEffect(Effect):
+    def __init__(self, context: 'Context', name: str,
+                 creator: Citizen) -> None:
+        super().__init__(context, name, creator, 13)
+
+    def _activate_impl(self) -> bool:
+        self.targets.append(self.creator)
+        self.user_interaction.save_active(msg.GangMessages.ACTIVATION_SUCCESS)
+
+        return True
+
+    def _resolve_impl(self) -> bool:
+        if self.game.round_number == self.activation_round:
+            return False
+
+        card_effects_list = []
+        citizens = self.context.game.citizens
+
+        for citizen in citizens:
+            card_effect = (effect for effect in citizen.effects
```

Comment:
Names are confusing.
card_effect is not a single effect but an iterator through bunch of effects.

And what is the difference between card_effect, effect and card_effect_list.
(While card_effect and card_effects_list are both iterable?)

Names should be reworked somehow